### DISCUSSION TITLE
Will Firecracker Continue to Run in Development Without Root Access?

### DISCUSSION THREAD

**Discussion by zicklag** [firecracker-microvm/firecracker#2767] [General]
In the design doc I read:

> In production environments, Firecracker should be started only via the jailer binary. The firecracker binary can also be executed directly, but this will no longer be possible in the future.

I just tried to run firecracker with jailer, but I ran into an "operation not permitted". Is that due to me missing root privileges on my machine? I'd just like it if in the future it was still possible to run firecracker without having to have root access. I love how it is able to run in userspace right now. :slightly_smiling_face: 

Also, maybe I misunderstood something and this won't be a problem at all.

**Comment by raduiliescu** [comment]
Hi @zicklag ,

Firecracker needs access to /dev/kvm and that is allowed only if a privileged user granted that permission. 
Most likely the "operation not permitted" error comes because the user launching Firecracker doesn't have /dev/kvm access.
Hope this clarifies your question.

Thanks!

**Reply by zicklag** [reply]
Hi @raduiliescu the interesting thing is that Firecracker works fine _without_ jailer.

Anyway, if it's nothing obvious, I won't worry about it too much, because my environment isn't the cleanest test environment and it's not a big deal to me now, but thanks for the reply!

**Reply by alsrdn** [reply]
@zicklag We have an issue on our ops board related to your question (https://github.com/firecracker-microvm/firecracker/issues/1190) . Is there a specific use-case in which you cannot use the jailer as root or do you have any concerns regarding running the jailer as root?
Although Firecracker can be run without the jailer, we don't recommend it for production environments.

**Reply by zicklag** [reply]
I just wanted to make sure that in the future I would be able to run firecracker without root, for development ( non production ) purposes.

Because the docs said that firecracker would eventually only run with `jailer`, and I couldn't get `jailer` to run without root, I just wanted to make sure that there would continue to be some way in which to run firecracker without root for development. This could be either firecracker running fine in dev without jailer, or having jailer run fine without root, as long as there will be a way to run firecracker dev without root.

I don't intend to run firecracker without jailer in production or without root, so the production case is fine for me.

**Comment by raduiliescu** [comment]
We have updated the documentation. Please see this PR - https://github.com/firecracker-microvm/firecracker/pull/2805

**Reply by zicklag** [reply]
Cool, thanks for the heads-up. :)
